sdk/src/trace/recordable.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace_api = opentelemetry::trace;
+namespace trace
+{
+/**
+ * Maintains a representation of a span in a format that can be processed by a recorder.
```

Comment:
Thread-compatible?

Reply:
This builds on top of https://github.com/open-telemetry/opentelemetry-cpp/pull/44.

But I didn't plan on making it thread-safe because the Span does the locking.
https://github.com/rnburn/opentelemetry-cpp/blob/recorder-interface/sdk/src/trace/span.h#L40

Reply:
Right. Could you please (probably in #44) document it as being thread-compatible (as opposed to thread-safe)

Reply:
Added a thread-compatible note to #44

Reply:
Thanks!